src/Boson_Sampling_Utilities.py

```diff
@@ -138,7 +138,13 @@ class EffectiveScatteringMatrixCalculator:
         In many methods of Boson Sampling simulations an effective scattering matrix has to be calculated. Therefore
         I decided to implement an calculator that'd be used if every single one of these methods.
     """
-    def __init__(self, matrix: ndarray, input_state: List[int] = [], output_state: List[int] = []):
+
+    def __init__(self, matrix: ndarray, input_state: Optional[List[int]] = None,
+                 output_state: Optional[List[int]] = None):
+        if output_state is None:
+            output_state = []
+        if input_state is None:
+            input_state = []
         self.__matrix = matrix
         self.__input_state = input_state
```

Comment:
Nitpick: this can be written like `self.__input_state = input_state if input_state is not None else []`.